### DISCUSSION TITLE
Distcc Dynamic host lists

### DISCUSSION THREAD

**Discussion by alw017** [distcc/distcc#528] [Q&A] [answered]
Is it possible to add or remove hosts from distcc's host list during the build? Is this even a good idea? 

Some context: I'm currently using Amazon EC2 instances to compile my code, so it would be best if they only ran when absolutely necessary. My intention is to let distcc scale dynamically for different build sizes (like requesting a new instance when there are more files to compile but no resources to do so) so I don't have to launch as many EC2 instances for smaller builds.

**Comment by sourcefrog** [comment, marked as answer]
I think what I would try there is putting an AWS loadbalancer in front of the instances (or even an auto-scaling group) and then tell distcc that the LB address is the one server it should use. (Let us know if that works!)

**Reply by alw017** [reply]
I did setup an network load balancer to handle distcc's requests, and it mostly works. There is one problem though:

It seems that some requests are getting dropped. Near the end of the end of the build, distcc gets hung up on a couple files that haven't been compiled even though distccmon-text shows no pending connection to the load balancer, and the entire process has wait for an IO timeout to retry. I'm not sure if this is an issue with my configuration of the load balancer or a limitation on distcc's part though.

**Reply by sourcefrog** [reply]
I don't know, it's hard to tell from here without more information.

**Reply by alw017** [reply]
I did some more testing, and it seems like it might have been an issue with auto scaling instances being too slow. Adding warm pool instances and reducing the scaling cooldown seems to have fixed the problem.
